internal/deploy/base.go

```diff
@@ -6,25 +6,33 @@ package deploy
 
 import (
 	"strings"
+
+	"github.com/elastic/e2e-testing/internal/installer"
 )
 
 // Deployment interface for operations dealing with deployments of the bits
 // required for testing
 type Deployment interface {
-	Add(services []string, env map[string]string) error    // adds a service to deployment
-	Bootstrap(waitCB func() error) error                   // will bootstrap or reuse existing cluster if kubernetes is selected
-	Destroy() error                                        // Teardown deployment
-	ExecIn(service string, cmd []string) (string, error)   // Execute arbitrary commands in service
-	Inspect(service string) (*ServiceManifest, error)      // inspects service
-	Remove(services []string, env map[string]string) error // Removes services from deployment
+	Add(services []string, env map[string]string) error                  // adds a service to deployment
+	AddFiles(service string, files []string) error                       // adds files to a service
+	Bootstrap(waitCB func() error) error                                 // will bootstrap or reuse existing cluster if kubernetes is selected
+	Destroy() error                                                      // Teardown deployment
+	ExecIn(service string, cmd []string) (string, error)                 // Execute arbitrary commands in service
+	Inspect(service string) (*ServiceManifest, error)                    // inspects service
+	Mount(service string, installType string) (installer.Package, error) // mounts a service for performing actions against it
+	Remove(services []string, env map[string]string) error               // Removes services from deployment
+	Start(service string) error                                          // Starts a service or container depending on Deployment
+	Stop(service string) error                                           // Stop a service or container depending on deployment
 }
 
 // ServiceManifest information about a service in a deployment
 type ServiceManifest struct {
```

Comment:
Do you think the ServiceRequest struct from #1162 will be shadowing this one? I was tempted to reuse this one when designing it, but the name `manifest` evoked me a response instead of a request.

Reply:
Yea let's use the servicerequest as much as we can I prefer yours over mine